src/ec/packages/ec-component-message/message.story.js

```diff
@@ -15,6 +19,40 @@ import dataWarning from './demo/data--warning';
 import message from './ecl-message.html.twig';
 import notes from './README.md';
 
+const iconsList = {};
+iconsList.none = null;
+
+notifIcons.forEach(icon => {
+  iconsList[icon] = icon;
+});
+
+const PrepareMessage = data => {
+  const name = select(
+    'icon.name',
+    iconsList,
+    data.icon.name,
+    buttonLabels.required
+  );
+  if (name !== null) {
+    getIconKnobs(data, name, 'notifications', 'l');
+  }
+
+  if (data.title) {
+    data.title = text('Title', data.title, buttonLabels.required);
+  }
+
+  if (data.description) {
+    data.description = text(
+      'Description',
+      data.description,
+      buttonLabels.required
+    );
+  }
+  getExtraKnobs(data);
+
+  return data;
```

Comment:
We should have also a knob for the variant, of type select, with the only option being the current one, since we have a story for each of them.

Also the thing you are now doing in formatIcon should actually be part of the main prepare function where you set them as knobs, still select with that only option, as mandatory, without these you don't render the component correctly otherwise.
This will also let you use in the json parameter of the notes the dataInfo, dataWhatever directly with no processing.